packages/google/src/google-generative-ai-embedding-model.ts

```diff
@@ -18,7 +18,7 @@ import {
 type GoogleGenerativeAIEmbeddingConfig = {
   provider: string;
   baseURL: string;
-  headers: () => Record<string, string | undefined>;
+  headers?: () => Promise<Record<string, string | undefined>>;
```

Comment:
this is a breaking change. we need to allow both variants (promise and non-promise). also it's a config that we use internally, so i recommend using `undefined` and not `?` (forces us to always set it)

Reply:
Made required, since it's not a promise it's easy to pass `{}` when we need to, so not adding an `| undefined`.